src/main/java/seedu/allonus/modules/StudyManager.java

```diff
@@ -106,6 +110,78 @@ public void deleteModule(String userInput) {
 
     }
 
+    public void editModule(String userInput, TextUi ui) {
+        try {
+            String moduleIndexString = userInput.replace("edit ", "");
+            int moduleIndex = Integer.parseInt(moduleIndexString) - 1;
+            if (modulesList.get(moduleIndex) != null) {
+                Module moduleToEdit = modulesList.get(moduleIndex);
+                printMessage("Here is the module that you have chosen to edit:");
+                printMessage(moduleToEdit.toString());
+                printMessage("Choose the part that you would like to edit: ");
+                boolean isEditFinished = false;
+                String editUserInput;
+                while (isEditFinished == false) {
+                    editUserInput = ui.getUserInput();
+                    if (editUserInput.startsWith("category")) {
+                        editModuleCategory(moduleToEdit, editUserInput);
+                        printMessage(moduleToEdit.toString());
+                    } else if (editUserInput.startsWith("code")) {
+                        editModuleCode(moduleToEdit, editUserInput);
+                        printMessage(moduleToEdit.toString());
+                    } else if (editUserInput.startsWith("day")) {
+                        editModuleDay(moduleToEdit, editUserInput);
+                        printMessage(moduleToEdit.toString());
+                    } else if (editUserInput.startsWith("time")) {
+                        editModuleTime(moduleToEdit, editUserInput);
+                        printMessage(moduleToEdit.toString());
+                    } else {
+                        printMessage("Your Module was successfully edited! Here are the changes");
+                        printMessage(moduleToEdit.toString());
+                        isEditFinished = true;
+                    }
```

Comment:
Is it possible to extract this portion as a method? This seems a little deeply nested.